### PR TITLE
Added S3_ENDPOINT variable

### LINKED ISSUE #3014: Support environment variable S3_ENDPOINT in tensorboard without tensorflow
When using tensorboard with tensorflow installed, we are able to set the s3_endpoint using S3_ENDPOINT environment variable. It is particularly usefull when running a local s3 instance like minio.

But when using the tensorboard package (without tensorflow), it uses the boto3 client, and don't use the S3_ENDPOINT variable.

A simple modification would be to replace boto3.client/session call with the same call but with the parameter: endpoint_url=os.environ.get('S3_ENDPOINT', None)

What do you think ?

### PR DESCRIPTION
**Pull request by thealphacod3r**
Fixes #3014 

@stephanwlee please review this!

### CODE REVIEW

**Comment by stephanwlee** [comment]
I've learned recently that TensorFlow generally prefer non-environment variable solutions. Any thoughts on this, @nfelt?

While this change looks reasonable, I think we actually should prefer the CLI flags to mirror AWS cli (https://docs.aws.amazon.com/cli/latest/reference/).

**Comment by thealphacod3r** [comment]
@nfelt please review this!

File: tensorboard/compat/tensorflow_stub/io/gfile.py [hunk 1]
```diff
@@ -220,6 +220,7 @@ class S3FileSystem(object):
     def __init__(self):
         if not boto3:
             raise ImportError("boto3 must be installed for S3 support.")
+        self.s3_endpoint = os.environ.get("S3_ENDPOINT", None)
```
Comment (nfelt) [inline tensorboard/compat/tensorflow_stub/io/gfile.py]: Please prefix with an underscore (`self._s3_endpoint`).

**Comment by thealphacod3r** [comment]
@googlebot I signed it!

File: tensorboard/compat/tensorflow_stub/io/gfile.py (hunk 1 above)
Comment (thealphacod3r) [reply tensorboard/compat/tensorflow_stub/io/gfile.py]: @nfelt i made the changes as you suggested please look into this, i ran black and reformatted the code but still the lint error exists!

**Comment by thealphacod3r** [comment]
@googlebot I consent.

**Comment by thealphacod3r** [comment]
@nfelt I recently changed my email address and now googlebot has marked my cla:no, i signed the cla with my new email address but still it's telling cla as no.

File: tensorboard/compat/tensorflow_stub/io/gfile.py (hunk 1 above)
Comment (nfelt) [reply tensorboard/compat/tensorflow_stub/io/gfile.py]: Can you rerun black?  As long as you're running it correctly it should ensure the lint passes on CI.

It looks like the problem is the spaces around the equals sign in the `endpoint_url = self._s3_endpoint` line below.

**Comment by thealphacod3r** [comment]
@nfelt should i close this and start with a new PR, it would be much easier to sort things then. Is it ok if i do so?

**Comment by nfelt** [comment]
It shouldn't be too hard to fix this PR, but if you prefer opening a new one, you can also do that.

**Pull request closed without merging**